src/App.js

```diff
@@ -16,25 +17,48 @@ class App extends Component {
 		};
 	}
 
-	componentWillMount() {
+	async componentWillMount() {
+        const promisePool = new PromisePool();
+        let promises;
+
+        /**
+		 * Each time one of the promises in the PromisePool
+		 * is resolved (or rejected), it will trigger this.
+         */
+        promisePool.subscribe((actionType, data) => {
+            if (actionType === 'settled') {
+                this.setState(state => {
+                    return {
+                        ...state,
+                        stats: state.stats.concat([data])
+                    };
+                })
+            } else if (actionType === 'finalized') {
+            	// they're all finished, let's clean up
+				promisePool.terminate();
+			}
+        });
+
 		if (window.location.search.indexOf('BINA') !== -1) {
 			console.log('BINANCE PLEASE');
 			this.setState({
 				exchange: 'Binance'
 			});
 
-			binaGetAccountBuilders().then(results => {
-				this.setState({
-					stats: results
-				});
-			});
+			promises = await binaGetAccountBuildersStreamable();
 		} else {
-			getAccountBuilders().then(results => {
-				this.setState({
-					stats: results
-				});
-			});
+            promises = await getAccountBuildersStreamable();
 		}
+
+        /**
+		 * set the initial container for the results
+         */
+        this.setState({ stats: [] });
```

Comment:
I would have done this in the initial state but it looks like the UI says there's no data if the `stats` array is ever an empty list, so we defer it until here.